__tests__/app.test.js

```diff
@@ -97,7 +97,33 @@ describe('/api/articles', () => {
       });
   });
 });
-
+describe('/api/articles?topic', () => {
+  test('GET 200: responds only with articles of a given topic', () => {
+    return request(app)
+      .get('/api/articles?topic=mitch')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles.length).toBe(12);
+        expect(body.articles[0].topic).toBe('mitch');
+      });
+  });
+  test('GET 400: sends an error when filtering by an invalid topic', () => {
+    return request(app)
+      .get('/api/articles?topic=not_mitch')
```

Comment:
This is a non-existent topic but the request is sound. If the topic is non-existent then this feels like a 404 response. There isn't really a bad request to send on this endpoint.